### PR TITLE
732 search

### PR DESCRIPTION
**Pull request by aledavila**
## Overview

Resolves 
https://github.ibm.com/carbon/issues/issues/732

Deprecated search with icons. Have been removed in react as well. 
https://github.com/carbon-design-system/carbon-components-react/pull/929

### Removed

Removed search with icons

### Changed

Changed close button offset to default 12px because Search with icons is deprecated.

### CODE REVIEW

**Comment by asudoh** [comment]
Thank you for working on this @aledavila! One question - Should large/small variants be identical with this change, or are there still some differences?

**Comment by aledavila** [comment]
@asudoh yes. they will be the same but on is smaller than the other.

**Review by asudoh — COMMENTED** [review]
Thanks @aledavila for clarifying! LGTM 👍 once the merge conflict is resolved.

**Review by tw15egan — APPROVED** [review]

**Review by marijohannessen — APPROVED** [review]

**Pull request merged**
